### DISCUSSION TITLE
Bulk import of domains

### DISCUSSION THREAD

**Discussion by akito2001** [deeztek/Hermes-Secure-Email-Gateway#64] [Support] [answered]
There is a script to bulk import recipients but we are a webhost for many customers and would like to look at hermes as an alternative for providing email to our customer domains.  We are currently using barracuda and looking for a way to bulk import our list of domains.  Has anyone been able to add domains programmatically?  I was able to follow the script at /admin/2/inc/create_new.cfm?type=domain to determine what tables needed to be added to inside of mysql to create the domain and the domain shows in the webui but that doesnt appear to be everything as hermes still wont accept mail for the newly created domain.  I've been looking through the other scripts under /admin to see if I could determine which other steps would be required.  

Any help or a point in the right direction is always appreciated.

**Comment by deeztek** [comment, marked as answer]
there are a lot of things that happen when a domain is added. The tables involved are domains, transport, senders in Hermes as well as a routine to add the domain in ciphermail tables which is not as simple as adding into Hermes. In addition there are routines to create various files for various subsystems. Bottom line, it's not a simple process and creating a single script to do that is not really feasible. That's the reason we don't have a field like in recipients to simply paste a large list of recipients at once.
